Submitted & accepted upstream in: https://github.com/mesonbuild/meson/pull/5917

From f91855fdde2348127e0bf6672e1a2e5d24601e09 Mon Sep 17 00:00:00 2001
From: Alan Coopersmith <alan.coopersmith@oracle.com>
Date: Sat, 29 Jun 2019 19:33:33 -0700
Subject: [PATCH] Fix "test cases/common/132 generated assembly" on Solaris

Without this change, the test fails with:

[11/12] Linking target square-gen-test.
warning: Text relocation remains                referenced
    against symbol                  offset      in file
square_unsigned                     0x15        square-gen-test@exe/main.c.o
[12/12] Linking target square-ct-test.
warning: Text relocation remains                referenced
    against symbol                  offset      in file
square_unsigned                     0x15        square-ct-test@exe/main.c.o

Signed-off-by: Alan Coopersmith <alan.coopersmith@oracle.com>
---
 test cases/common/132 generated assembly/square-x86_64.S.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test cases/common/132 generated assembly/square-x86_64.S.in b/test cases/common/132 generated assembly/square-x86_64.S.in
index 856af137..b2cf3ebe 100644
--- a/test cases/common/132 generated assembly/square-x86_64.S.in	
+++ b/test cases/common/132 generated assembly/square-x86_64.S.in	
@@ -19,7 +19,7 @@ END
 .text
 .globl SYMBOL_NAME(square_unsigned)
 /* Only supported with GAS */
-# if defined(__linux__) || defined(__DragonFly__) || defined(__FreeBSD__) || defined(__NetBSD__)
+# if defined(__linux__) || defined(__DragonFly__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__sun)
 .type  square_unsigned,@function
 # endif
 
-- 
2.19.2

